File: c++/1021.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

int main() {

   double n, cent;
   cin >> n;

   cout << "NOTAS:" << endl;
   cout << int(n) / 100 << " nota(s) de R$ 100.00" << endl;
   cout << int(n) % 100 / 50 << " nota(s) de R$ 50.00" << endl;
   cout << (int(n) % 100) % 50 / 20 << " nota(s) de R$ 20.00" << endl;
   cout << ((int(n) % 100) % 50) % 20 / 10 << " nota(s) de R$ 10.00" << endl;
   cout << (((int(n) % 100) % 50) % 20) % 10 / 5 << " nota(s) de R$ 5.00" << endl;
   cout << ((((int(n) % 100) % 50) % 20) % 10) % 5 / 2 << " nota(s) de R$ 2.00" << endl;
   cout << "MOEDAS:" << endl;
   cout << (((((int(n) % 100) % 50) % 20) % 10) % 5) % 2 << " moeda(s) de R$ 1.00" << endl;
   cent = int(((n - (int)n) * 100));
   cout << int(cent) % 100 / 50 << " moeda(s) de R$ 0.50" << endl;
   cout << (int(cent) % 100) % 50 / 25 << " moeda(s) de R$ 0.25" << endl;
   cout << ((int(cent) % 100) % 50) % 25 / 10 << " moeda(s) de R$ 0.10" << endl;
   cout << (((int(cent) % 100) % 50) % 25) % 10 / 5 << " moeda(s) de R$ 0.05" << endl;
   cout << ((((int(cent) % 100) % 50) % 20) % 10) % 5 << " moeda(s) de R$ 0.01" << endl;

   return 0;
}

File: python3/1021.py
entrada = float(input())

nota = int(entrada)
moeda = int((entrada - nota) * 100)

print("NOTAS:")
print("{} nota(s) de R$ 100.00".format(nota // 100))
nota = nota % 100
print("{} nota(s) de R$ 50.00".format(nota // 50))
nota = nota % 50
print("{} nota(s) de R$ 20.00".format(nota // 20))
nota = nota % 20
print("{} nota(s) de R$ 10.00".format(nota // 10))
nota = nota % 10
print("{} nota(s) de R$ 5.00".format(nota // 5))
nota = nota % 5
print("{} nota(s) de R$ 2.00".format(nota // 2))
nota = nota % 2

print("MOEDAS:")
print("{} moeda(s) de R$ 1.00".format(nota))
moeda = moeda % 100
print("{} moeda(s) de R$ 0.50".format(moeda // 50))
moeda = moeda % 50
print("{} moeda(s) de R$ 0.25".format(moeda // 25))
moeda = moeda % 25
print("{} moeda(s) de R$ 0.10".format(moeda // 10))
moeda %= 10
print("{} moeda(s) de R$ 0.05".format(moeda // 5))
moeda %= 5
print("{} moeda(s) de R$ 0.01".format(moeda))
